mne/beamformer/_compute_beamformer.py

```diff
@@ -44,93 +43,67 @@ def _check_src_type(filters):
     return filters, warn_text
 
 
-def _prepare_beamformer_input(info, forward, label, picks, pick_ori,
-                              fwd_norm=None, apply_proj=True):
-    """Input preparation common for all beamformer functions.
+def _prepare_beamformer_input(info, forward, label=None, pick_ori=None,
+                              noise_cov=None, rank=None, pca=False, loose=None,
+                              combine_xyz='fro', exp=None, limit=None,
+                              allow_fixed_depth=True, limit_depth_chs=False):
+    """Input preparation common for LCMV, DICS, and RAP-MUSIC."""
+    _check_option('pick_ori', pick_ori,
+                  ('normal', 'max-power', 'vector', None))
 
-    Check input values, prepare channel list and gain matrix. For documentation
-    of parameters, please refer to _apply_lcmv.
-    """
-    is_free_ori = forward['source_ori'] == FIFF.FIFFV_MNE_FREE_ORI
-
-    if pick_ori in ['normal', 'max-power', 'vector']:
-        if not is_free_ori:
-            raise ValueError(
-                'Normal or max-power orientation can only be picked '
-                'when a forward operator with free orientation is used.')
-    elif pick_ori is not None:
-        raise ValueError('pick_ori must be one of "normal", "max-power", '
-                         '"vector", or None, got %s' % (pick_ori,))
+    # Restrict forward solution to selected vertices
+    if label is not None:
+        _, src_sel = label_src_vertno_sel(label, forward['src'])
+        forward = _restrict_forward_to_src_sel(forward, src_sel)
+
+    if loose is None:
+        loose = 0. if is_fixed_orient(forward) else 1.
+    if noise_cov is None:
+        noise_cov = make_ad_hoc_cov(info, std=1.)
+    forward, info_picked, gain, _, orient_prior, _, trace_GRGT, noise_cov, \
+        whitener = _prepare_forward(
+            forward, info, noise_cov, 'auto', loose, rank=rank, pca=pca,
+            use_cps=True, exp=exp, limit_depth_chs=limit_depth_chs,
+            combine_xyz=combine_xyz, limit=limit,
+            allow_fixed_depth=allow_fixed_depth)
+    is_free_ori = not is_fixed_orient(forward)  # could have been changed
+    nn = forward['source_nn']
+    if is_free_ori:  # take Z coordinate
+        nn = nn[2::3]
+    nn = nn.copy()
+    vertno = _get_vertno(forward['src'])
+    if forward['surf_ori']:
+        nn[...] = [0, 0, 1]  # align to local +Z coordinate
```

Comment:
Is this not taken care of in the ``fwd`` ?

Reply:
Not always I don't think (depends on `surf_ori` somehow IIRC). This is copy-paste from before and when I removed it something broke IIRC